azurerm/resource_arm_servicebus_subscription_rule.go

```diff
@@ -7,14 +7,15 @@ import (
 	"github.com/Azure/azure-sdk-for-go/services/servicebus/mgmt/2017-04-01/servicebus"
 	"github.com/hashicorp/terraform/helper/schema"
 	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/response"
 	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
 )
 
 func resourceArmServiceBusSubscriptionRule() *schema.Resource {
 	return &schema.Resource{
-		Create: resourceArmServiceBusSubscriptionRuleCreate,
+		Create: resourceArmServiceBusSubscriptionRuleCreateUpdate,
```

Comment:
I don't know if dual-purpose create/update functions are a good idea. An edge case I could see in the fact that ID is set in both cases is that if the ID changes for some reason (that is not expected), it will result in an orphaned resource. I definitely understand that there is a lot of common functionality between both workflows (possibly nearly all of it), but I think it's important to differentiate - the shared functionality could be wrapped, for example.

Reply:
the Azure API's are (almost) all upserts - thus we're calling the same SDK method/API endpoint with the same data, as such splitting this out into separate Update calls is probably unnecessary imo.

Regarding the ID potentially changing, since we're retrieving the ID from the `Get` endpoint, as opposed to the Create/Update endpoint, which could be different - these should be consistent; given this is a low likelihood based on the way the Azure API design works - I'd suggest instead we can patch any relevant bugs with, should this affect any endpoints:

```
if d.IsNewResource() {
  resp, err := client.Get(resourceGroup, name)
  if err != nil {
    return fmt.Errorf("Error retrieving ID for [Name] %q (Resource Group %q): %+v)", resourceGroup, name, err)
  }
  d.SetId(*resp.ID)
}
```

WDYT?

Reply:
This all sounds sane. My comment was not implying that this should be held back :slightly_smiling_face: I did also notice though that other Azure resources were not sharing the same kind of upserting convention in the CRUD functions, at least in naming. Still LGTM!